src/predicate/str/difference.rs

```diff
@@ -0,0 +1,69 @@
+use difference;
+
+use Predicate;
+
+/// Predicate that diffs two strings.
+///
+/// This is created by the `predicate::str::diff`.
+#[derive(Clone, Debug)]
+pub struct DifferencePredicate {
+    orig: String,
+    split: String,
+    distance: i32,
+}
+
+impl DifferencePredicate {
+    /// The split used when identifying changes.
+    ///
+    /// Common splits include:
+    /// - `""` for char-level.
+    /// - `" "` for word-level.
+    /// - `"\n" for line-level.
+    ///
+    /// Default: `"\n"`
+    pub fn split<S>(mut self, split: S) -> Self
```

Comment:
Instead of forcing a potential allocation, this could use the copy-on-write `Cow` type

Reply:
I'm always on the fence on allocation vs indirection (`Cow`).  I'll go ahead and switch it to `Cow`.